src/OrchardCore.Modules/OrchardCore.Roles/Services/RoleStore.cs

```diff
@@ -115,7 +116,7 @@ public async Task<IRole> FindByIdAsync(string roleId, CancellationToken cancella
             // While updating find a role from the loaded document being mutated.
             var roles = _updating ? await LoadRolesAsync() : await GetRolesAsync();
 
-            var role = roles.Roles.FirstOrDefault(x => x.RoleName == roleId);
+            var role = roles.Roles.FirstOrDefault(x => string.Equals(x.RoleName, roleId, StringComparison.OrdinalIgnoreCase));
```

Comment:
Yes, it alters behavior, but enhances the reliability of the RoleStore. The comparison of roles by name was handled inconsistently across the entire source code. If anyone used multiple roles that differed only in casing, they are already facing issues.

The specific problem I encountered was that default permissions were not applied to roles in `UpdateAsync` due to the case sensitivity. Merely making `UpdateAsync` case-insensitive wouldn't have been a logical solution.

Reply:
What should be done is storing the role name in the database as normalized and always use `==` to compare normalize values. I don't think we should use invariant check all over the place

Reply:
The normalized value is already stored in `NormalizedRoleName` which is used by MS Identity. But I don't think we gain anything by using that. It would always require to either normalize all parameters or also using case insensitive comparing. The normalized value is also managed by MS Identity and not by OC on its own.
As it is currently implemented, `RoleName` is used as the ID for roles and it is a user-provided value. That makes it kind of fragile and I think case insensitive comparing is the least intrusive solution.

Reply:
@gvkries 
I don't remember all the details, I would have to look at all the places we use role name to provide more feedback. Which is why I suggested to replace string with constants in this PR. 

But from what I remember is that we do not store normalized roles in the Users properties. I think we need to change how to store the role name and add migration to update existing stored names. Then, anytime we do comparisons, we would normalize the name and do a normalized name comparison.